lib/nolibc/stdio.c

```diff
@@ -541,3 +543,78 @@ int puts(const char *s)
 {
 	return fputs_internal(s, stdout, 1);
 }
+
+#if CONFIG_LIBVFSCORE
+
```

Comment:
Here you can define the `strcut _nolibc_file`. For now, it is probably just:
```c
struct _nolibc_file {
    int fd;
    bool eof; /* true if we reached end-of-file */
};
```
Typically, you would also add a buffer and additional fields that are needed to reduce syscall overheads but we can start minimal since we want a simple wrapper for `nolibc` only.